students/kate/session04/trigrams.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+"""
+
+trigrams with sherlock writing
+
+"""
+# import random module
+from random import randint
+
+# function to make pairs and followers in trigram dictionary
+def gen_trigrams(words):
+    # make an empty dicitonary for the trigrams
+    trigrams_dict = {}
+    for i in range(len(words)-2):
+        pair = tuple(words[i:i+2])
+        follower = words[i+2]
+        print(pair, follower)
+        if pair in trigrams_dict:
+            trigrams_dict[pair].append(follower)
+        trigrams_dict[pair] = [follower]
```

Comment:
careful! this should be in an else block.

this way, you are always writing over the existing pair, even if it was there.

Reply:
oh, and you can use dict.setdefault to reduce it to one line.